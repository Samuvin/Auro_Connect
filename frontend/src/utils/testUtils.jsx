import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Create a custom render function that includes providers
export const renderWithProviders = (ui, options = {}) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  const AllTheProviders = ({ children }) => {
    return (
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

// Mock user data for testing
export const mockUser = {
  _id: '1',
  name: 'John Doe',
  username: 'johndoe',
  email: 'john@example.com',
  profilePicture: '/avatar.png',
  headline: 'Software Developer',
  connections: ['2', '3'],
  location: 'New York, NY'
};

// Mock post data for testing
export const mockPost = {
  _id: '1',
  content: 'This is a test post',
  author: mockUser,
  likes: ['2', '3'],
  comments: [
    {
      _id: '1',
      content: 'Great post!',
      user: mockUser,
      createdAt: new Date().toISOString()
    }
  ],
  image: 'https://example.com/image.jpg',
  createdAt: new Date().toISOString()
}; 