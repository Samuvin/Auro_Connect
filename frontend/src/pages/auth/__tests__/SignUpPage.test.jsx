import { screen } from '@testing-library/react';
import SignUpPage from '../SignUpPage';
import { renderWithProviders } from '../../../utils/testUtils';

describe('SignUpPage', () => {
  test('renders signup page correctly', () => {
    renderWithProviders(<SignUpPage />);
    
    expect(screen.getByText('Make the most of your professional life')).toBeInTheDocument();
  });

  test('contains signup form', () => {
    renderWithProviders(<SignUpPage />);
    
    expect(screen.getByPlaceholderText('Full name')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Username')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Password (6+ characters)')).toBeInTheDocument();
  });

  test('has link to login page', () => {
    renderWithProviders(<SignUpPage />);
    
    const loginLink = screen.getByRole('link', { name: /sign in/i });
    expect(loginLink).toBeInTheDocument();
  });

  // Snapshot Tests
  test('matches snapshot for signup page layout', () => {
    const { container } = renderWithProviders(<SignUpPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for signup page form elements', () => {
    const { container } = renderWithProviders(<SignUpPage />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 