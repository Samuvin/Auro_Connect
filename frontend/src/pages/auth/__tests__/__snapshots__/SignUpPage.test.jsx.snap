// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignUpPage matches snapshot for signup page form elements 1`] = `
<div
  class="min-h-screen flex flex-col justify-center py-12 sm:px-6 lg:px-8"
>
  <div
    class="sm:mx-auto sm:w-full sm:max-w-md"
  >
    <div
      class="flex justify-center items-center w-full px-2 mb-6"
    >
      <svg
        class="h-20 w-auto text-blue-600"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M287.6 54.2c-10.8-2.2-22.1-3.3-33.5-3.6V32.4c78.1 2.2 146.1 44 184.6 106.6l-15.8 9.1c-6.1-9.7-12.7-18.8-20.2-27.1l-18 15.5c-26-29.6-61.4-50.7-101.9-58.4l4.8-23.9zM53.4 322.4l23-7.7c-6.4-18.3-10-38.2-10-58.7s3.3-40.4 9.7-58.7l-22.7-7.7c3.6-10.8 8.3-21.3 13.6-31l-15.8-9.1C34 181 24.1 217.5 24.1 256s10 75 27.1 106.6l15.8-9.1c-5.3-10-9.7-20.3-13.6-31.1zM213.1 434c-40.4-8-75.8-29.1-101.9-58.7l-18 15.8c-7.5-8.6-14.4-17.7-20.2-27.4l-16 9.4c38.5 62.3 106.8 104.3 184.9 106.6v-18.3c-11.3-.3-22.7-1.7-33.5-3.6l4.7-23.8zM93.3 120.9l18 15.5c26-29.6 61.4-50.7 101.9-58.4l-4.7-23.8c10.8-2.2 22.1-3.3 33.5-3.6V32.4C163.9 34.6 95.9 76.4 57.4 139l15.8 9.1c6-9.7 12.6-18.9 20.1-27.2zm309.4 270.2l-18-15.8c-26 29.6-61.4 50.7-101.9 58.7l4.7 23.8c-10.8 1.9-22.1 3.3-33.5 3.6v18.3c78.1-2.2 146.4-44.3 184.9-106.6l-16.1-9.4c-5.7 9.7-12.6 18.8-20.1 27.4zM496 256c0 137-111 248-248 248S0 393 0 256 111 8 248 8s248 111 248 248zm-12.2 0c0-130.1-105.7-235.8-235.8-235.8S12.2 125.9 12.2 256 117.9 491.8 248 491.8 483.8 386.1 483.8 256zm-39-106.6l-15.8 9.1c5.3 9.7 10 20.2 13.6 31l-22.7 7.7c6.4 18.3 9.7 38.2 9.7 58.7s-3.6 40.4-10 58.7l23 7.7c-3.9 10.8-8.3 21-13.6 31l15.8 9.1C462 331 471.9 294.5 471.9 256s-9.9-75-27.1-106.6zm-183 177.7c16.3-3.3 30.4-11.6 40.7-23.5l51.2 44.8c11.9-13.6 21.3-29.3 27.1-46.8l-64.2-22.1c2.5-7.5 3.9-15.2 3.9-23.5s-1.4-16.1-3.9-23.5l64.5-22.1c-6.1-17.4-15.5-33.2-27.4-46.8l-51.2 44.8c-10.2-11.9-24.4-20.5-40.7-23.8l13.3-66.4c-8.6-1.9-17.7-2.8-27.1-2.8-9.4 0-18.5.8-27.1 2.8l13.3 66.4c-16.3 3.3-30.4 11.9-40.7 23.8l-51.2-44.8c-11.9 13.6-21.3 29.3-27.4 46.8l64.5 22.1c-2.5 7.5-3.9 15.2-3.9 23.5s1.4 16.1 3.9 23.5l-64.2 22.1c5.8 17.4 15.2 33.2 27.1 46.8l51.2-44.8c10.2 11.9 24.4 20.2 40.7 23.5l-13.3 66.7c8.6 1.7 17.7 2.8 27.1 2.8 9.4 0 18.5-1.1 27.1-2.8l-13.3-66.7z"
        />
      </svg>
      <h2
        class="text-3xl font-bold text-gray-900 ml-2"
      >
        Auto_Connect
      </h2>
    </div>
    <h2
      class="text-center text-3xl font-extrabold text-gray-900"
    >
      Make the most of your professional life
    </h2>
  </div>
  <div
    class="mt-8 sm:mx-auto sm:w-full sm:max-w-md shadow-md"
  >
    <div
      class="bg-white py-8 px-4 shadow sm:rounded-lg sm:px-10"
    >
      <form
        class="flex flex-col gap-4"
      >
        <input
          class="input input-bordered w-full"
          placeholder="Full name"
          required=""
          type="text"
          value=""
        />
        <input
          class="input input-bordered w-full"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
        <input
          class="input input-bordered w-full"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
        <input
          class="input input-bordered w-full"
          placeholder="Password (6+ characters)"
          required=""
          type="password"
          value=""
        />
        <button
          class="btn btn-primary w-full text-white"
          type="submit"
        >
          Agree & Join
        </button>
      </form>
      <div
        class="mt-6"
      >
        <div
          class="relative"
        >
          <div
            class="absolute inset-0 flex items-center"
          >
            <div
              class="w-full border-t border-gray-300"
            />
          </div>
          <div
            class="relative flex justify-center text-sm"
          >
            <span
              class="px-2 bg-white text-gray-500"
            >
              Already on LinkedIn?
            </span>
          </div>
        </div>
        <div
          class="mt-6"
        >
          <a
            class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-blue-600 bg-white hover:bg-gray-50"
            href="/login"
          >
            Sign in
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SignUpPage matches snapshot for signup page layout 1`] = `
<div
  class="min-h-screen flex flex-col justify-center py-12 sm:px-6 lg:px-8"
>
  <div
    class="sm:mx-auto sm:w-full sm:max-w-md"
  >
    <div
      class="flex justify-center items-center w-full px-2 mb-6"
    >
      <svg
        class="h-20 w-auto text-blue-600"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M287.6 54.2c-10.8-2.2-22.1-3.3-33.5-3.6V32.4c78.1 2.2 146.1 44 184.6 106.6l-15.8 9.1c-6.1-9.7-12.7-18.8-20.2-27.1l-18 15.5c-26-29.6-61.4-50.7-101.9-58.4l4.8-23.9zM53.4 322.4l23-7.7c-6.4-18.3-10-38.2-10-58.7s3.3-40.4 9.7-58.7l-22.7-7.7c3.6-10.8 8.3-21.3 13.6-31l-15.8-9.1C34 181 24.1 217.5 24.1 256s10 75 27.1 106.6l15.8-9.1c-5.3-10-9.7-20.3-13.6-31.1zM213.1 434c-40.4-8-75.8-29.1-101.9-58.7l-18 15.8c-7.5-8.6-14.4-17.7-20.2-27.4l-16 9.4c38.5 62.3 106.8 104.3 184.9 106.6v-18.3c-11.3-.3-22.7-1.7-33.5-3.6l4.7-23.8zM93.3 120.9l18 15.5c26-29.6 61.4-50.7 101.9-58.4l-4.7-23.8c10.8-2.2 22.1-3.3 33.5-3.6V32.4C163.9 34.6 95.9 76.4 57.4 139l15.8 9.1c6-9.7 12.6-18.9 20.1-27.2zm309.4 270.2l-18-15.8c-26 29.6-61.4 50.7-101.9 58.7l4.7 23.8c-10.8 1.9-22.1 3.3-33.5 3.6v18.3c78.1-2.2 146.4-44.3 184.9-106.6l-16.1-9.4c-5.7 9.7-12.6 18.8-20.1 27.4zM496 256c0 137-111 248-248 248S0 393 0 256 111 8 248 8s248 111 248 248zm-12.2 0c0-130.1-105.7-235.8-235.8-235.8S12.2 125.9 12.2 256 117.9 491.8 248 491.8 483.8 386.1 483.8 256zm-39-106.6l-15.8 9.1c5.3 9.7 10 20.2 13.6 31l-22.7 7.7c6.4 18.3 9.7 38.2 9.7 58.7s-3.6 40.4-10 58.7l23 7.7c-3.9 10.8-8.3 21-13.6 31l15.8 9.1C462 331 471.9 294.5 471.9 256s-9.9-75-27.1-106.6zm-183 177.7c16.3-3.3 30.4-11.6 40.7-23.5l51.2 44.8c11.9-13.6 21.3-29.3 27.1-46.8l-64.2-22.1c2.5-7.5 3.9-15.2 3.9-23.5s-1.4-16.1-3.9-23.5l64.5-22.1c-6.1-17.4-15.5-33.2-27.4-46.8l-51.2 44.8c-10.2-11.9-24.4-20.5-40.7-23.8l13.3-66.4c-8.6-1.9-17.7-2.8-27.1-2.8-9.4 0-18.5.8-27.1 2.8l13.3 66.4c-16.3 3.3-30.4 11.9-40.7 23.8l-51.2-44.8c-11.9 13.6-21.3 29.3-27.4 46.8l64.5 22.1c-2.5 7.5-3.9 15.2-3.9 23.5s1.4 16.1 3.9 23.5l-64.2 22.1c5.8 17.4 15.2 33.2 27.1 46.8l51.2-44.8c10.2 11.9 24.4 20.2 40.7 23.5l-13.3 66.7c8.6 1.7 17.7 2.8 27.1 2.8 9.4 0 18.5-1.1 27.1-2.8l-13.3-66.7z"
        />
      </svg>
      <h2
        class="text-3xl font-bold text-gray-900 ml-2"
      >
        Auto_Connect
      </h2>
    </div>
    <h2
      class="text-center text-3xl font-extrabold text-gray-900"
    >
      Make the most of your professional life
    </h2>
  </div>
  <div
    class="mt-8 sm:mx-auto sm:w-full sm:max-w-md shadow-md"
  >
    <div
      class="bg-white py-8 px-4 shadow sm:rounded-lg sm:px-10"
    >
      <form
        class="flex flex-col gap-4"
      >
        <input
          class="input input-bordered w-full"
          placeholder="Full name"
          required=""
          type="text"
          value=""
        />
        <input
          class="input input-bordered w-full"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
        <input
          class="input input-bordered w-full"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
        <input
          class="input input-bordered w-full"
          placeholder="Password (6+ characters)"
          required=""
          type="password"
          value=""
        />
        <button
          class="btn btn-primary w-full text-white"
          type="submit"
        >
          Agree & Join
        </button>
      </form>
      <div
        class="mt-6"
      >
        <div
          class="relative"
        >
          <div
            class="absolute inset-0 flex items-center"
          >
            <div
              class="w-full border-t border-gray-300"
            />
          </div>
          <div
            class="relative flex justify-center text-sm"
          >
            <span
              class="px-2 bg-white text-gray-500"
            >
              Already on LinkedIn?
            </span>
          </div>
        </div>
        <div
          class="mt-6"
        >
          <a
            class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-blue-600 bg-white hover:bg-gray-50"
            href="/login"
          >
            Sign in
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
